action/copy.go

```diff
@@ -193,6 +200,72 @@ func CopyNotSeekableFileToRemote(r io.Reader, objectKey string) (total int64, er
 	return total, nil
 }
 
+// CopySeekableFileToRemote will copy a seekable file to remote.
+func CopySeekableFileToRemote(r ReadAtCloseSeeker, objectKey string) (total int64, err error) {
+	defer r.Close()
+	dataLen, err := r.Seek(0, io.SeekEnd)
+	if err != nil {
+		log.Errorf("Get file size failed [%v]", err)
+		return 0, constants.ErrorFileSizeInvalid
+	}
+	uploadID, err := contexts.Storage.InitiateMultipartUpload(objectKey)
+	if err != nil {
+		return
+	}
+
+	log.Debugf("Object <%s> uploading via upload ID <%s>", objectKey, uploadID)
+
+	partSize, partCount, err := CalculatePartSizeAndCount(dataLen)
+	if err != nil {
+		return
+	}
+
+	_, err = r.Seek(0, io.SeekStart)
+	if err != nil {
+		return 0, err
+	}
+	var wg sync.WaitGroup
+	var uploadErrFlag bool
+	for partNumber := 0; partNumber < partCount; partNumber++ {
+		wg.Add(1)
+		go func(r ReadAtCloseSeeker, partNumber int) {
```

Comment:
We need to control the goroutines because user may update files larger then 10TB. This could be done in another PR.